modules/product/src/main/java/com/opengamma/strata/product/SecurityId.java

```diff
@@ -0,0 +1,210 @@
+   * If necessary, the scheme can be encoded using {@link StandardId#encodeScheme(String)}.
+   * <p>
+   * The value must be non-empty and match the regular expression '{@code [!-z][ -z]*}'.
+   *
+   * @param scheme  the scheme of the identifier, not empty
+   * @param value  the value of the identifier, not empty
+   * @return the security identifier
+   * @throws IllegalArgumentException if the scheme or value is invalid
+   */
+  public static SecurityId of(String scheme, String value) {
+    return of(StandardId.of(scheme, value));
+  }
+
+  /**
+   * Creates an instance from a standard two-part identifier.
+   *
+   * @param standardId  the underlying standard two-part identifier
+   * @return the security identifier
+   */
+  public static SecurityId of(StandardId standardId) {
+    return new SecurityId(standardId);
+  }
+
+  /**
+   * Parses an {@code StandardId} from a formatted scheme and value.
+   * <p>
+   * This parses the identifier from the form produced by {@code toString()}
+   * which is '{@code $scheme~$value}'.
+   *
+   * @param str  the identifier to parse
+   * @return the security identifier
+   * @throws IllegalArgumentException if the identifier cannot be parsed
+   */
+  @FromString
+  public static SecurityId parse(String str) {
+    return new SecurityId(StandardId.parse(str));
+  }
+
+  // creates an identifier for a combined calendar
+  private SecurityId(StandardId standardId) {
+    this.standardId = ArgChecker.notNull(standardId, "standardId");
+  }
+
+  // resolve after deserialization
+  private Object readResolve() {
+    return of(standardId);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Gets the standard two-part identifier.
+   * 
+   * @return the standard two-part identifier
+   */
+  public StandardId getStandardId() {
+    return standardId;
+  }
+
+  /**
+   * Gets the type of data this identifier refers to.
+   * <p>
+   * A {@code SecurityId} refers to a {@code ReferenceSecurity}.
+   *
+   * @return the type of the reference data this identifier refers to
+   */
+  @Override
+  public Class<ReferenceSecurity> getReferenceDataType() {
+    return ReferenceSecurity.class;
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Resolves this identifier to a security using the specified reference data.
+   * <p>
+   * This returns an instance of {@link ReferenceSecurity}.
+   * <p>
+   * Resolved objects may be bound to data that changes over time.
+   * If the data changes, the resolved form will not be updated.
+   * Care must be taken when placing the resolved form in a cache or persistence layer.
+   * 
+   * @param refData  the reference data, used to resolve the reference
+   * @return the resolved security
+   * @throws ReferenceDataNotFoundException if the identifier is not found
+   */
+  @Override
+  public ReferenceSecurity resolve(ReferenceData refData) {
```

Comment:
Maybe those should be brought into line too. Why have two ways to achieve the exact same thing? It adds confusion and inconsistency without any upside I can think of.